python/agents/Plumber-Data-Engineering-Assistant/plumber_agent/sub_agents/monitoring_agent/tools/dataproc.py

```diff
@@ -0,0 +1,272 @@
+from google.cloud import logging_v2
+from datetime import datetime, timedelta, timezone
+import subprocess
+import logging
+
+logger = logging.getLogger("plumber-agent")
+
+
+def get_dataproc_cluster_logs_with_name(project_id: str, dataproc_cluster_name: str = "", _limit: int = 10) -> dict:
+    
+    """ 
+    Fetches Google Cloud Logging log entries specifically for Dataproc clusters. 
+
+    This function retrieves log entries from Google Cloud Logging for a specified
+    Dataproc cluster with it's name, with an option to limit the number of results. 
+    It is designed to provide an overview of recent cluster activities and potential issues.
+
+    Args:
+        project_id (str, required): The Google Cloud project ID. from which logs are to be fetched.
+        dataproc_cluster_name (str, required): The name of the Dataproc cluster
+            to filter logs for. If not provided don't call this tool.
+        _limit (int, optional): The maximum number of log entries to retrieve.
+            The function will fetch up to this many entries, Defaults to 10.
+
+    Returns:
+        dict: A dictionary containing the status of the operation and the retrieved logs.
+            The dictionary will have the following structure:
+                - "status" (str): "success" if logs were fetched successfully, or "error"
+                    if an error occurred.
+                - "report" (str): A human-readable message detailing the outcome.
+                    If successful, it includes a summary and a list of the fetched log entries.
+                    If no logs are found, it indicates that.
+                - "message" (str, optional): Only present if "status" is "error", providing
+                    details about the error that occurred.
+
+    Note: [IMPORTANT]
+        - Call this tool only when user want's logs of a cluster with the name 
+    """
+
+    try:
+        collected_logs = get_dataproc_logs(project_id=project_id, resource_name="cloud_dataproc_cluster", query_str=dataproc_cluster_name, filter="cluster_name",_limit=_limit)
+        log_count = len(collected_logs)
+        if log_count == 0:
+            print("No log entries found matching the criteria.")
```

Comment:
Let's replace all the prints in this file with logger (or just remove them)

Reply:
this is taken care in latest commit